sdk/core/core-client/src/serviceClient.ts

```diff
@@ -88,7 +97,11 @@ export class ServiceClient {
    */
   constructor(options: ServiceClientOptions = {}) {
     this._requestContentType = options.requestContentType;
-    this._baseUri = options.baseUri;
+    this._endpoint = options.endpoint ?? options.baseUri;
+    if (options.baseUri) {
+      const logger = createClientLogger("ServiceClientOptions");
```

Comment:
instead of making a one-off logger each time this happens, can we set up a package-wide logger for `core-client` similar to how we did for `core-rest-pipeline`: https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/core/core-rest-pipeline/src/log.ts

Reply:
I add a log.ts in core-client here https://github.com/Azure/azure-sdk-for-js/pull/21125/files#diff-d97379932c5dafaf7552f5c08d259cee242df0ffeadb85be4daab382a073261b, but also notice that here https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/core/core-client/src/authorizeRequestOnClaimChallenge.ts#L9 we have a create a logger directly from @azure/logger. Should we do something with it?

Reply:
Very good question! Can you update the code in `authorizeRequestOnClaimChallenge.ts` to use the same shared `core-client` logger you just added?

Reply:
@xirzec updated. Can you help take a look again ? Thanks